File: app/src/main/java/hk/collaction/contentfarmblocker/ui/adapter/AppItemAdapter.java
package hk.collaction.contentfarmblocker.ui.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import hk.collaction.contentfarmblocker.R;
import hk.collaction.contentfarmblocker.model.AppItem;

/**
 * Created by himphen on 25/5/16.
 */
public class AppItemAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

	private List<AppItem> mDataList;
	private ItemClickListener mListener;

	public interface ItemClickListener {
		void onItemDetailClick(AppItem appItem);
	}

	public AppItemAdapter(List<AppItem> mDataList, ItemClickListener mListener) {
		this.mDataList = mDataList;
		this.mListener = mListener;
	}

	@Override
	public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
		Context mContext = parent.getContext();

		View itemView = LayoutInflater.from(mContext).inflate(R.layout.list_item_info_app, parent, false);
		return new ItemViewHolder(itemView);
	}

	@Override
	public void onBindViewHolder(RecyclerView.ViewHolder rawHolder, int position) {
		AppItem item = mDataList.get(position);
		ItemViewHolder holder = (ItemViewHolder) rawHolder;

		holder.titleTv.setText(item.getAppName());
		holder.contentTv.setText(item.getPackageName());
		holder.iconIv.setImageDrawable(item.getIcon());

		holder.rootView.setTag(item);
		holder.rootView.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View view) {
				mListener.onItemDetailClick((AppItem) view.getTag());
			}
		});
	}

	@Override
	public int getItemCount() {
		return mDataList == null ? 0 : mDataList.size();
	}

	static class ItemViewHolder extends RecyclerView.ViewHolder {

		@BindView(R.id.text1)
		TextView titleTv;
		@BindView(R.id.text2)
		TextView contentTv;
		@BindView(R.id.icon)
		ImageView iconIv;
		@BindView(R.id.root_view)
		LinearLayout rootView;

		ItemViewHolder(View itemView) {
			super(itemView);
			ButterKnife.bind(this, itemView);
		}
	}
}

File: app/src/main/java/hk/collaction/contentfarmblocker/ui/adapter/AppItemAdapter.kt
package hk.collaction.contentfarmblocker.ui.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import hk.collaction.contentfarmblocker.R
import hk.collaction.contentfarmblocker.model.AppItem

/**
 * Created by himphen on 25/5/16.
 */
class AppItemAdapter(
        private val dataList: List<AppItem>,
        private val listener: ItemClickListener
) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    interface ItemClickListener {
        fun onItemDetailClick(appItem: AppItem)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return ItemViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.list_item_info_app, parent, false))
    }

    override fun onBindViewHolder(rawHolder: RecyclerView.ViewHolder, position: Int) {
        val item = dataList[position]
        val holder = rawHolder as ItemViewHolder
        holder.titleTv.text = item.appName
        holder.contentTv.text = item.packageName
        holder.iconIv.setImageDrawable(item.icon)
        holder.rootView.tag = item
        holder.rootView.setOnClickListener { view -> listener.onItemDetailClick(view.tag as AppItem) }
    }

    override fun getItemCount(): Int = dataList.size

    internal class ItemViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var titleTv: TextView = view.findViewById(R.id.text1)
        var contentTv: TextView = view.findViewById(R.id.text2)
        var iconIv: ImageView = view.findViewById(R.id.icon)
        var rootView: LinearLayout = view.findViewById(R.id.root_view)
    }

}